x/lockup/keeper/lock.go

```diff
@@ -455,6 +456,38 @@ func (k Keeper) unlockMaturedLockInternalLogic(ctx sdk.Context, lock types.Perio
 	return nil
 }
 
+// SetLockRewardReceiverAddress changes the reward recipient address to the given address.
+// Storing an empty string for reward receiver would indicate the owner being reward receiver.
+func (k Keeper) SetLockRewardReceiverAddress(ctx sdk.Context, lockID uint64, owner sdk.AccAddress, newReceiverAddress string) error {
+	lock, err := k.GetLockByID(ctx, lockID)
+	if err != nil {
+		return err
+	}
+
+	// check if the lock owner is the method caller.
+	if lock.GetOwner() != owner.String() {
+		return types.ErrNotLockOwner
+	}
+
+	// if the given receiver address is same as the lock owner, we store an empty string instead.
+	if lock.Owner == newReceiverAddress {
+		newReceiverAddress = ""
```

Comment:
Can we define "" as a placeholder variable in `lock.types`?

```go
const defaultOwnerReceiverPlaceholder = ""
```

Reply:
What is the reason we cannot just set it to the lock owner's address?

Reply:
Addressed in https://github.com/osmosis-labs/osmosis/pull/5281/commits/80b11f5b33ad87b407ed876e98ece45f4bfa6191, further convo about using lock owner address directly going on online